pkgs/applications/editors/emacs/default.nix

```diff
@@ -67,11 +61,17 @@ stdenv.mkDerivation rec {
              "--with-gif=no" "--with-tiff=no" ])
     ++ lib.optional withXwidgets "--with-xwidgets";
 
-  NIX_CFLAGS_COMPILE = lib.optionalString (stdenv.isDarwin && withX)
-    "-I${cairo.dev}/include/cairo";
-
-  preBuild = ''
+  preConfigure = lib.optionalString srcRepo ''
+    ./autogen.sh
+  '' + lib.optionalString srcLocal ''
```

Comment:
I find having steps meant for working from the git repository mixed with the "regular" steps (i.e. meant to be run from a proper tarball) confusing. This makes the purpose of said steps harder to decipher after the fact, and requires doing a lot of git archeology to clean them up.

I think that the srcRepo option makes sense, as you can point to a git rev, set that option and expect the whole compilation to go through.

Regarding the srcLocal one, I can see two courses of action:

1. Getting rid of it: As it currently stands, it provides only a very incomplete set of (one) steps. The C objects files are not cleaned, etc.
2. Keeping it: It would be really nice to have a more robust set of steps allowing you to use this derivation for emacs development. Maybe using something based off of https://github.com/jwiegley/nix-config/blob/master/emacsHEAD.nix

At this stage, I am for removing srcLocal and the associated step, and revisit the issue in the future.